### PR TITLE
Pretrained issue

### PR DESCRIPTION
**Pull request by Joao-L-S-Almeida**
It automatically sets `pretrained=False ` when no config is available (as for Swin).

### CODE REVIEW

**Comment by CarlosGomes98** [comment]
I think its safer to throw an error here and force the user to specify `pretrained=False`. What we can do is have the error message ask the user to do just that. In this way, the user is crystal clear on what weights area actually being loaded (or not) in the model.

**Comment by romeokienzler** [comment]
@Michal-Muszynski can you please explains why the config file can't be changed in the process?

**Comment by romeokienzler** [comment]
@Michal-Muszynski what is the reason you can't update the config file?

**Comment by biancazadrozny** [comment]
Need to test if during inference this error occurs when using a fine-tuned model and the backbone is not there, with this flag.

**Comment by romeokienzler** [comment]
@romeokienzler please review

File: terratorch/models/backbones/prithvi_vit.py [hunk 1]
```diff
@@ -85,15 +85,29 @@ def _create_prithvi(
...
+            pretrained_strict=True,
+            encoder_only=encoder_only,
+            **kwargs,
+        )
+    except RuntimeError:
+        print(f"No pretrained configuration was found for the model {variant}.")
+        model = build_model_with_cfg(
+            TemporalViTEncoder,
+            variant,
+            False,
```
Comment (romeokienzler) [inline terratorch/models/backbones/prithvi_vit.py:105]: here it seems to be correct, pse correct above @Joao-L-S-Almeida

**Comment by romeokienzler** [comment]
@Joao-L-S-Almeida I think there is one little problem, there rest lgtm thanks  a lot!

File: terratorch/models/backbones/prithvi_vit.py (hunk 1 above)
Comment (Joao-L-S-Almeida) [reply terratorch/models/backbones/prithvi_vit.py:105]: The difference is that the exception imposes `pretrained=False`:
```
    try:
        model = build_model_with_cfg(
            TemporalViTEncoder,
            variant,
            pretrained,
            pretrained_filter_fn=checkpoint_filter_wrapper_fn,
            pretrained_strict=True,
            encoder_only=encoder_only,
            **kwargs,
        )
    except RuntimeError:
        print(f"No pretrained configuration was found for the model {variant}.")
        model = build_model_with_cfg(
            TemporalViTEncoder,
            variant,
            False,
            pretrained_filter_fn=checkpoint_filter_wrapper_fn,
            pretrained_strict=True,
            encoder_only=encoder_only,
            **kwargs,
        )
```

File: terratorch/models/backbones/prithvi_swin.py [hunk 2]
```diff
@@ -177,15 +177,30 @@ def _create_swin_mmseg_transformer(
...
+            pretrained_strict=False,
+            feature_cfg={"flatten_sequential": True, "out_indices": out_indices},
+            **kwargs,
+        )
+    except RuntimeError:
+        print(f"No pretrained configuration was found for the model {variant}.")
+        model: MMSegSwinTransformer = build_model_with_cfg(
+            MMSegSwinTransformer,
+            variant,
+            False,
```
[the remark this answers is not in the document]
Comment (romeokienzler) [reply terratorch/models/backbones/prithvi_swin.py:197]: @Joao-L-S-Almeida this one definitely needs to be fixed, then we can merge

File: terratorch/models/backbones/prithvi_vit.py (hunk 1 above)
Comment (Joao-L-S-Almeida) [reply terratorch/models/backbones/prithvi_vit.py:105]: The same here (terratorch/models/backbones/prithvi_swin.py):
    # When the pretrained configuration is not available in HF, we shift to 
    # pretrained=False
    try:
        model: MMSegSwinTransformer = build_model_with_cfg(
            MMSegSwinTransformer,
            variant,
            pretrained,
            pretrained_filter_fn=checkpoint_filter_wrapper_fn,
            pretrained_strict=False,
            feature_cfg={"flatten_sequential": True, "out_indices": out_indices},
            **kwargs,
        )
    except RuntimeError:
        print(f"No pretrained configuration was found for the model {variant}.")
        model: MMSegSwinTransformer = build_model_with_cfg(
            MMSegSwinTransformer,
            variant,
            False,
            pretrained_filter_fn=checkpoint_filter_wrapper_fn,
            pretrained_strict=False,
            feature_cfg={"flatten_sequential": True, "out_indices": out_indices},
            **kwargs,
        )

Comment (Joao-L-S-Almeida) [reply terratorch/models/backbones/prithvi_vit.py:105]: Is it answering the question, @romeokienzler ?

**Review by romeokienzler — APPROVED** [review]

**Pull request merged**